shadow.c

```diff
@@ -549,15 +559,29 @@ static char *get_full_path(const char *filename TSRMLS_DC)
 		SHADOW_G(curdir) = getcwd(NULL, 0);
 	}
 
+#if PHP_VERSION_ID < 50600
 	new_state.cwd = strdup(SHADOW_G(curdir));
+#else
+	new_state.cwd = estrdup(SHADOW_G(curdir));
```

Comment:
Out of curiosity can we use the same combination of either (strdup/free or estrdup/efree) regardless of the version?